.buildkite/pipeline.yml

```diff
@@ -1,10 +1,26 @@
-use_plugins: &use_plugins
+x_use_plugins: &use_plugins
     plugins:
       - automattic/bash-cache#v1.0.0: ~
       - automattic/git-s3-cache#add/vm-host-support:
           bucket: "a8c-repo-mirrors"
           repo: "wordpress-mobile/wordpress-ios/"
 
+# Common environment values for the `env` key.
+#
+# To add more, simply add another entry at the same indentation as `IMAGE_ID`.
+#
+# Usage in a step with only these values:
+#
+#   env: *common_env
+#
+# Usage in a step with other values:
+#
+#   env:
+#     <<: *common_env
+#     other_key: "other value"
+x_common_env: &common_env
+  IMAGE_ID: xcode-12.5.1
```

Comment:
I this approach in this commit because I like how it forces each step to explicitly define the `env` key. If our pipeline files grow in size, a reader won't have to jump to the definition of the anchor to know it's used for the `env` key, which they would have had to do with the `<<: *common_env` approach.


```yml
env: *common_env
# vs
<<: *common_env_with_env_root_in_it
```

I also decided to use underscores instead of the suggested dashes to keep it in line with the other anchor already defined (`use_plugin`) and the Buildkite [spec](buildkite/pipeline-schema@f7a0894/schema.json) which uses underscores, e.g. in `artifacts_paths`.